### PR TITLE
Issue #293 - Add view for viewing Firefox Interventions

### PR DESCRIPTION
**Pull request by magsout**
Screen Shot 2020-06-09 at 14 32 10
Screen Shot 2020-06-09 at 14 32 25

### CODE REVIEW

**Comment by magsout** [comment]
@denschub it's better to just draw one line but display in the tooltip each number?

**Comment by denschub** [comment]
No, not really. Is there a reason not to use Chart.js stacked line chart? That pretty much looks exactly what's perfect here: https://www.chartjs.org/samples/latest/charts/area/line-stacked.html - individual lines with individual numbers per line, but the y-axis representing the total number of all targets combined.

**Comment by magsout** [comment]
>No, not really. Is there a reason not to use Chart.js stacked line chart? 

this is what i use. But the dataset is not correct and yes, I can use this option.

**Comment by denschub** [comment]
> But the dataset is not correct

What do you mean by that? :) The data in https://webcompat-metrics.herokuapp.com/data/firefox-interventions?distribution=mozilla-central&type=ua_override&start=2020-06-08, for example, is accurate. There are two overrides for all platforms, 21 for android, four for desktop, which makes a total of 27, which is correct. Do you need me to change something in the way I return the data?

**Comment by magsout** [comment]
@denschub ah, no sorry. My dataset I mean ☺️

**Comment by denschub** [comment]
Ahh, okay. :) Communication is hard, heh. Let me know if you need any help, I'm also happy to chat on Slack or Zoom!

**Comment by magsout** [comment]
@denschub 

>Writing this, I actually noticed this works fine on hard-reload, but not if you change the filters without hard reloading. The graph adjust it's axis scale, but the drawn data appears to be the same. Is there some caching going on?

fixed in my last PR. I didn't reset the object so every time we fetch the API,  the data was concat...

**Comment by magsout** [comment]
I added the config, but it doesn't work. I try to figured out why.

```
yAxes: [{
  stacked: true,
}]
```

**Comment by denschub** [comment]
Looking at the current preview-deployment on heroku, things appear to be fine:

Screenshot 2020-06-09 at 22 09 51

We have a total of 53 interventions across all platforms (8+35+10) and that's where the top line is on the Y-axis! Looks perfect to  me, are you seeing something different?

**Comment by magsout** [comment]
I'm seeing the same thing. maybe I'm wrong. but in the legend we see three colors for each platform. On the graph we see 3 curves, except that the red curve shows all of the 3 curves. But what about the last curve? In this case `affecting desktop`.

**Comment by magsout** [comment]
but if it's ok for you, we can merge.

**Comment by denschub** [comment]
I think I see what's confusing you: the fact that this is a stacked line graph, not a graph with three individual lines. Each of the lines "adds" to the line below, hence why there is a mismatch between what the label and the Y-axis says. However, this is a valuable type of graph, as it allows you to easily visually see the total number of interventions, without having to calculate the sum in your head.

However, usually, those kinds of charts have their area colored in, see https://datavizcatalogue.com/methods/stacked_area_graph.html and also https://www.chartjs.org/samples/latest/charts/area/line-stacked.html. If you look at these charts, is it more obvious what they are doing there?

**Comment by magsout** [comment]
@denschub thanks. I get it now. 

>https://datavizcatalogue.com/methods/stacked_area_graph.html 

very good.

>However, usually, those kinds of charts have their area colored in, see 

as you mentionned this, I cand do that:
Screen Shot 2020-06-12 at 09 40 47

**Comment by denschub** [comment]
> as you mentioned this, I could do that:

let's do it! I think it makes the graph easier to understand, and having a stacked line graph filled in is pretty standard. :)

File: src/constants/Charts.js [hunk 1]
```diff
@@ -7,3 +7,44 @@ export const CHART_LINE = "CHART_LINE";
...
+ * Interventions
+ */
+
+export const INTERVENTION_COUNTER_ALL_VALUE = "all";
+export const INTERVENTION_COUNTER_ANDROID_VALUE = "android";
+export const INTERVENTION_COUNTER_DESKTOP_VALUE = "desktop";
+
+export const INTERVENTION_COUNTER_LIST = {
+  [INTERVENTION_COUNTER_ALL_VALUE]: {
+    label: "All platforms",
```
Comment (denschub) [inline src/constants/Charts.js:21]: While you're on changing the area color, can you call this "Universal"? :)

File: src/containers/Intervention/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            scales: {
+              xAxes: [
+                {
+                  scaleLabel: {
+                    display: true,
+                  },
+                  type: "time",
+                  distribution: "linear",
+                  time: {
+                    minUnit: "month",
```
Comment (magsout) [inline src/containers/Intervention/index.js:63]: the unit is month, but if it's not enough we can edit and set by date.

**Comment by magsout** [comment]
- added color area
- updated label by universal

File: src/containers/Intervention/index.js (hunk 2 above)
Comment (denschub) [reply src/containers/Intervention/index.js:63]: That's perfect, we only really work in it once per release cycle, so `month` works fine.

**Comment by magsout** [comment]
thanks a lot

**Pull request merged**